src/components/Search/SearchHeader.tsx

```diff
@@ -0,0 +1,90 @@
+import { useSearchStore } from '~/components/Search/useSearchState';
+import { useInstantSearch, useSearchBox } from 'react-instantsearch-hooks-web';
+import {
+  createStyles,
+  Group,
+  SegmentedControl,
+  SegmentedControlItem,
+  Text,
+  ThemeIcon,
+  Title,
+} from '@mantine/core';
+import { IconCategory, IconFileText } from '@tabler/icons-react';
+import { useRouter } from 'next/router';
+import { removeEmpty } from '~/utils/object-helpers';
+
+const useStyles = createStyles((theme) => ({}));
+
+export const SearchHeader = () => {
+  const { uiState } = useInstantSearch();
+  const { setSearchParamsByUiState, ...states } = useSearchStore((state) => state);
+  const { theme } = useStyles();
+  const [index] = Object.keys(uiState);
+  const { query } = useSearchBox();
+  const router = useRouter();
+
+  const onChangeIndex = (value: string) => {
+    setSearchParamsByUiState(uiState);
+
+    if (states.hasOwnProperty(value)) {
+      // Redirect to the route with the relevant state:
+      router.replace(
+        {
+          pathname: `/search/${value}`,
+          query: removeEmpty({
+            ...states[value as keyof typeof states],
+            query: query || null, // Remove empty string from URL
+          }),
+        },
+        undefined,
+        { shallow: true }
+      );
+    } else {
+      router.replace(`/search/${value}`);
+    }
+  };
+
+  const data: SegmentedControlItem[] = [
+    {
+      label: (
+        <Group align="center" spacing={8} noWrap>
+          <ThemeIcon
+            size={30}
+            color={index === 'models' ? theme.colors.dark[7] : 'transparent'}
+            p={6}
```

Comment:
Based on the designs, I believe these should be rounded
```suggestion
            p={6}
            radius="xl"
```